### PR TITLE
Deformable convolution v8: ngraph part

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
- *spec changes:

   1.  bilinear_interpolation_padding attribute type `int` -> `boolean`. It is not entirely clear how to use this attribute with `int` type, it comes down to checks like bilinear_interpolation_padding == 0, bilinear_interpolation_padding == 1
   2. renamed ModulationScalars to Mask, Modulation Scalars is inconvenient to use.

 - *ngraph implementation of DeformableConvolution v8*
 - *new base class for DeformableConvolution v1 and v8*
 - *type prop unit tests*

### Tickets:
 - *56579*

other parts (python API, reference implementation, MO, transformations) will be submitted as separate PRs after this PR is merged. (https://github.com/openvinotoolkit/openvino/pull/6313)

### CODE REVIEW

**Review by apertovs — APPROVED** [review]

File: ngraph/core/src/op/util/deformable_convolution_base.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    , m_auto_pad(auto_pad)
+    , m_group(group)
+    , m_deformable_group(deformable_group)
+{
+}
+
+bool op::util::DeformableConvolutionBase::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(util_DeformableConvolutionBase_visit_attributes);
+    visitor.on_attribute("strides", m_strides);
```
Comment (ilyachur) [inline ngraph/core/src/op/util/deformable_convolution_base.cpp:40]: Can we add tests for visitor API?

Comment (itikhono) [reply ngraph/core/src/op/util/deformable_convolution_base.cpp:40]: done

**Review by GlebKazantaev — APPROVED** [review]

File: ngraph/core/include/ngraph/opsets/opset8_tbl.hpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ NGRAPH_OP(ConvolutionBackpropData, ngraph::op::v1)
 NGRAPH_OP(Cos, ngraph::op::v0)
 NGRAPH_OP(Cosh, ngraph::op::v0)
 NGRAPH_OP(CumSum, ngraph::op::v0)
-NGRAPH_OP(DeformableConvolution, ngraph::op::v1)
+NGRAPH_OP(DeformableConvolution, ngraph::op::v8)
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/opsets/opset8_tbl.hpp:32]: Please add new operations to the end of file.

**Review by ilyachur — APPROVED** [review]

File: ngraph/core/include/ngraph/opsets/opset8_tbl.hpp (hunk 2 above)
Comment (itikhono) [reply ngraph/core/include/ngraph/opsets/opset8_tbl.hpp]: done

**Review by lazarevevgeny — APPROVED** [review]
Spec looks good.

**Pull request merged**